### PR TITLE
All countries on one page

### PR DESCRIPTION
**Pull request by lukipuki**
It's slow as hell but if that's what the users want :smile: Screenshot_20200420_082953

Will try caching the graphs next.

### CODE REVIEW

File: covid_graphs/covid_graphs/scripts/country_dashboard.py [hunk 1]
```diff
@@ -13,11 +14,54 @@
...
+            value="Normal",
+            labelStyle={"display": "inline-block", "margin": "0 4px 0 0"},
+        ),
+    )
+    return buttons
+
+
 def create_graphs(
-    reports: List[CountryReport], prediction_event: PredictionEvent,
+    report_by_name: Dict[str, CountryReport], prediction_event: PredictionEvent,
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/country_dashboard.py]: Nit: report_by_short_name

File: covid_graphs/covid_graphs/scripts/country_dashboard.py [hunk 2]
```diff
@@ -13,11 +14,54 @@
...
 ):
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/country_dashboard.py]: Nit: You can add a return type annotation:
```
) -> List[CountryGraph]:
```

File: covid_graphs/covid_graphs/scripts/country_dashboard.py [hunk 3]
```diff
@@ -27,11 +71,7 @@ def create_graphs(
     return country_graphs
 
 
-def create_dashboard(
-    data_dir: Path, server: Flask,
-):
-    # TODO(miskosz): Don't use print.
-    print("Creating dashboard for prediction graphs.")
+def _prepare_data_structures(data_dir: Path):
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/country_dashboard.py]: Here I would add a description of the return value, at least as a type annotation:
```
-> Tuple[Dict[str, PredictionEvent], Dict[str, List[CountryGraph]]]
```

Pro tip: You can define a type alias somewhere in the global scope at the top of the file (below imports), e.g.:
```
CountryGraphsByReportName = Dict[str, List[CountryGraph]]
```

File: covid_graphs/covid_graphs/scripts/server.py [hunk 4]
```diff
@@ -53,17 +53,26 @@ def covid19_redirect():
 
 
 def _run_flask_server(server: Flask, data_dir: Path):
-    _create_prediction_app(server=server, data_dir=data_dir)
+    _create_prediction_apps(server=server, data_dir=data_dir)
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/server.py:60]: I admire your attention to detail 😄

File: covid_graphs/covid_graphs/scripts/server.py [hunk 5]
```diff
@@ -33,7 +33,7 @@ def home():
 
     @server.route("/covid19/normal")
     def covid19_redirect():
-        return redirect(url_for("covid19_predictions"))
+        return redirect(url_for("covid19_all_predictions"))
```
Comment (miskosz) [inline covid_graphs/covid_graphs/scripts/server.py:36]: I think single predictions should be the default (i.e. the one that does not lag).

We should also redirect from `/covid19/predictions` now. (Maybe you can even have two `@server.route` decorators for one function?)

**Review by miskosz — APPROVED** [review]
LGTM. It got tricky to share functionality between `create_[single|all]_countr[y|ies]_dashboard`. It is good for now, we can refactor later.

File: covid_graphs/covid_graphs/scripts/country_dashboard.py (hunk 1 above)
Comment (lukipuki) [reply covid_graphs/covid_graphs/scripts/country_dashboard.py]: Done

File: covid_graphs/covid_graphs/scripts/country_dashboard.py (hunk 3 above)
Comment (lukipuki) [reply covid_graphs/covid_graphs/scripts/country_dashboard.py]: Done

File: covid_graphs/covid_graphs/scripts/server.py (hunk 5 above)
Comment (lukipuki) [reply covid_graphs/covid_graphs/scripts/server.py:36]: I disagree, the old URL should go to the dashboard with the old functionality - "Do not break users" :smile: Yes, we might eat too much of their memory, but this consistency seems to be more important to them (the few that I talked to).

`/covid19/predictions/` can go to `/single/`, though.

File: covid_graphs/covid_graphs/scripts/server.py (hunk 4 above)
Comment (lukipuki) [reply covid_graphs/covid_graphs/scripts/server.py:60]: My weapon and my curse :wink:

File: covid_graphs/covid_graphs/scripts/server.py (hunk 5 above)
Comment (miskosz) [reply covid_graphs/covid_graphs/scripts/server.py:36]: Ok then either is good. Your call for the default for the `/covid19/predictions/`.

**Pull request merged**